lgc/builder/BuilderRecorder.cpp

```diff
@@ -346,7 +346,7 @@ ShaderModes *BuilderRecorder::getShaderModes() {
 // @param instName : Name to give instruction(s)
 Value *BuilderRecorder::CreateDotProduct(Value *const vector1, Value *const vector2, const Twine &instName) {
   Type *const scalarType = cast<VectorType>(vector1->getType())->getElementType();
-  return record(Opcode::DotProduct, scalarType, {vector1, vector2}, instName);
+  return record(Opcode::DotProduct, scalarType, {vector1, vector2}, instName, Attribute::ReadNone);
```

Comment:
What I prefer is to add call attribute within the function record() if the attributes are not provided, something like this:
```
record() {
  ...
  // Add default attributes if they are not set
  if (attibute.empty()) {
    switch (opcode) {
    case X:
    case Y:
      attribute.push_back();
    }
  }
}
```
So you don't need to change so many codes and similar opcode could be handled in a group. I just propose this. It depends on Tim's judgement.